packages/core/factories/src/useLocaleFactory.ts

```diff
@@ -0,0 +1,88 @@
+import {computed, Ref, ref} from '@vue/composition-api';
+import { UseLocale } from '@vue-storefront/interfaces';
+
+export type UseLocaleFactoryParams<
+  LOCALE,
+  COUNTRY,
+  CURRENCY,
+  AVAILABLE_LOCALES = LOCALE[],
+  AVAILABLE_COUNTRIES = COUNTRY[],
+  AVAILABLE_CURRENCIES = CURRENCY[]> = {
```

Comment:
we'll use agnostic types for this, so:
* removing additional types here,
* export methods for loading available lists separately
* adding params for loading methods for filters (for example for fetching currencies only for a specific country)

Reply:
the part about loading requires https://github.com/DivanteLtd/next/issues/274